estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,62 @@
+from odoo import models, fields, api
+from datetime import timedelta
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Property offer"
+    _order = "price desc"
+
+    price = fields.Float(required=True)
+    status = fields.Selection(selection = [("accepted","Accepted"),("refused","Refused")])
+    partner_id = fields.Many2one("res.partner", string = "Partner", required=True)
+    property_id = fields.Many2one("estate.property",string = "Property", required = True)
+
+
+
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+
+    
+ 
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+
+    @api.depends("validity","create_date")
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = record.create_date + timedelta(days=record.validity) 
+            else:
+                record.date_deadline = fields.Date.today() + timedelta(days=record.validity) 
+    
+    def _inverse_date_deadline(self):
+        for record in self:
+            if record.date_deadline and record.create_date: 
+                delta = record.date_deadline - record.create_date.date()
+                record.validity = delta.days
+            else:
+                record.validity = (record.date_deadline - fields.Date.today()).days
+
+
+
+    def action_accept(self):
+        for record in self:
+            if record.property_id.estate == "sold":
+                raise UserError("Cannot accept offeer for a sold property")
+            record.status == "accepted"
+            record.property_id.buyer  = record.partner_id
+            record.property_id.estate = "offer accepted"
+            record.property_id.selling_price  = record.price
+        return True
+
+
+    def action_refuse(self):
+        for record in self:
+           record.status = "refused"
+        return True
+
+    _sql_constraints = [
+        ('check_price_positive', 'CHECK(price >= 0)',
+        'The offer price of should be greater than 0.'),      
+    ]
```

Comment:
sql constraints are ussually at the top of the model, right after the fields, just so that we can see them faster